src/deploy/functions/tasks.ts

```diff
@@ -120,13 +120,10 @@ export function createFunctionTask(
       operationResourceName: op.name,
       onPoll,
     });
-    if (!backend.isEventTrigger(fn.trigger)) {
+    if (!backend.isEventTrigger(fn.trigger) && fn.invoker) {
```

Comment:
Honestly, on second review, I think we should possibly have a version of setInvoker that does blind writes vs one that updates on top of an existing IAM policy. It'd be nice to cut N IAM policy changes out, but it isn't a blocker.

Reply:
I've set `fn.invoker = ["public"]` if it's undefined and added a check in the if statement `fn.invoker[0] !== "private"`. Since in the sdk we already make sure that `public` and `private` are in an array of length 1, checking the first element here makes sense to me. Thoughts?

Also, I created two setInvoker functions `setInvokerCreate` and `setInvokerUpdate`. `setInvokerCreate` does not pull the current IAM policies, just assigns the invoker role and corresponding members. `setInvokerUpdate` does the GET call first, then updates the invoker role members.